internal_transfers/actions/tests/e2e/getInternalImbalance.spec.ts

```diff
@@ -0,0 +1,157 @@
+import { getInternalImbalance, MinimalTxData } from "../../src/accounting";
+import { TenderlySimulator } from "../../src/simulate/tenderly";
+import { ethers } from "ethers";
+
+const simulator = new TenderlySimulator(
+  process.env["TENDERLY_USER"] || "INVALID_USER",
+  process.env["TENDERLY_PROJECT"] || "TENDERLY_PROJECT",
+  process.env["TENDERLY_ACCESS_KEY"] || "TENDERLY_ACCESS_KEY"
+);
+
+async function TxDataFromHash(txHash: string): Promise<MinimalTxData> {
+  const provider = ethers.getDefaultProvider(
+    process.env["NODE_URL"] || "NODE_URL"
+  );
+  const transaction = await provider.getTransactionReceipt(txHash);
+  if (transaction === null) {
+    throw new Error(`invalid transaction hash ${txHash} - try again`);
+  }
+  const { from, hash, logs } = transaction;
+  const tenderlyLogs = logs.map((log) => {
+    const { address, data, topics } = log;
+    return {
+      address,
+      data,
+      // had to Map it to make a copy of a readonly field.
+      topics: topics.map((value) => value),
+    };
+  });
```

Comment:
Nit, feel free to ignore:

```suggestion
  const tenderlyLogs = logs.map((log) => ({
    ...log,
    // had to Map it to make a copy of a readonly field.
    topics: log.topics.map((value) => value),
  }));
```